src/lib.rs

```diff
@@ -1324,8 +1329,14 @@ pub mod sync {
         /// assert_eq!(Lazy::force_mut(&mut lazy), &mut 92);
         /// ```
         pub fn force_mut(this: &mut Lazy<T, F>) -> &mut T {
-            Self::force(this);
-            Self::get_mut(this).unwrap_or_else(|| unreachable!())
+            if this.cell.get_mut().is_none() {
+                let value = match this.init.get_mut().take() {
+                    Some(f) => f(),
+                    None => panic!("Lazy instance has previously been poisoned"),
+                };
+                this.cell = OnceCell::with_value(value);
+            }
+            this.cell.get_mut().unwrap_or_else(|| unreachable!())
```

Comment:
(Rewriting this so as to avoiding the double-`get_mut()` on the happy path [runs into `polonius`](https://docs.rs/polonius-the-crab/0.3.1/polonius_the_crab/index.html#rationale-limitations-of-the-nll-borrow-checker))